### ISSUE TITLE
Make database port adjustable

### ISSUE THREAD

**Issue by Superdanby** [DOMjudge/domjudge-packaging#37]
It is inconvenient and pretty confusing when we can set the port for judges to connect to the server, but couldn't do the same between the database and the server.

Related files: https://github.com/DOMjudge/domjudge-packaging/search?q=MYSQL_HOST&unscoped_q=MYSQL_HOST

Using the [`-P` option](https://linux.die.net/man/1/mysqlshow) may solve the problem.

**Comment by nickygerritsen** [comment]
Unfortunately currently domjudge itself does support custom ports yet in the `etc/dbpasswords.secret` file.
Note that this is legacy code so I will check where it is still used, so we can at least track this.

**Comment by thijskh** [comment]
So it does mean that only some commandline scripts would not work. But the main components of DOMjudge would?

**Comment by nickygerritsen** [comment]
That’s what I think but I’m not sure.

**Comment by thijskh** [comment]
In that case let’s put the work into upgrading those scripts DOMjudge/domjudge#425

**Comment by nickygerritsen** [comment]
I think this can now be done since we only use the secrets file in `dj_setup_database` and in `load_db_secrets.php`

**Comment by thijskh** [comment]
An even more radical approach would be to no longer use the dbpasswords.secret intermediate format but write out .env.local content directly... not sure if we would gain much at this point though

**Comment by nickygerritsen** [comment]
That was me who was still logged in with the wrong account

**Comment by thijskh** [comment]
Parse = include since it’s just shell

**Comment by nickygerritsen** [comment]
But i don’t think the `mysql` binary (which we use to create stuff) understands the env bar, does it?

**Comment by thijskh** [comment]
Use separate variables, like in this example? https://symfony.com/doc/current/configuration.html#env-file-syntax

**Comment by nickygerritsen** [comment]
I wonder if that is a best practice. I feel like symfony is promoting DATABASE_URL. But even if they do, shouldn’t be too hard to parse it?
